### DISCUSSION TITLE
Issue with Multipart Form Data

### DISCUSSION THREAD

**Discussion by elacoursiere** [adnanh/webhook#509] [Q&A] [answered]
Hi,

I try to setup a hook for [Plex Media Server](https://github.com/adnanh/webhook/blob/master/docs/Hook-Examples.md#multipart-form-data), but I get:
[webhook] 2021/02/12 03:14:58 Completed 404 Not Found

parse-parameters-as-json seems to be at the root cause, but I can't find out how to set it properly.

I get _Hook not found._ in response to an http post request.

Can you help?

Thanks

**Comment by moorereason** [comment, marked as answer]
It looks like you're not hitting the right URL.  Can you run webhook with verbose logging and share that?  Also, can you share your hooks configuration?

**Reply by elacoursiere** [reply]
Hi Cameron!

Config in Plex:
http://my_dev_server:9000/hooks/plex

Log:
[webhook] 2021/02/12 17:42:26 version 2.5.0 starting
[webhook] 2021/02/12 17:42:26 setting up os signal watcher
[webhook] 2021/02/12 17:42:26 attempting to load hooks from /home/elacoursiere/webhook/hooks.json
[webhook] 2021/02/12 17:42:26 found 1 hook(s) in file
[webhook] 2021/02/12 17:42:26 	loaded: plex
[webhook] 2021/02/12 17:42:26 serving hooks on http://0.0.0.0:9000/hooks/{id}
[webhook] 2021/02/12 17:42:26 os signal watcher ready
[webhook] 2021/02/12 17:42:41 Started POST /hooks/plex
[webhook] 2021/02/12 17:42:41 plex got matched
[webhook] 2021/02/12 17:42:41 error parsing form payload invalid URL escape "%@\xc6"
[webhook] 2021/02/12 17:42:41 error parsing JSON parameters: couldn't retrieve argument for {Source:payload Name:payload EnvName:}
[webhook] 2021/02/12 17:42:41 Completed 400 Bad Request in 9.857125ms
[webhook] 2021/02/12 17:42:43 Started POST /hooks/plex
[webhook] 2021/02/12 17:42:43 plex got matched
[webhook] 2021/02/12 17:42:43 error parsing JSON parameters: couldn't retrieve argument for {Source:payload Name:payload EnvName:}
[webhook] 2021/02/12 17:42:43 Completed 400 Bad Request in 649.453µs

My Hooks config is a copy-paste from the doc : 
```
[
  {
    "id": "plex",
    "execute-command": "/home/elacoursiere/webhook/webhook_plex.sh",
    "parse-parameters-as-json": [
      {
        "source": "payload",
        "name": "payload"
      }
    ],
    "trigger-rule":
    {
      "match":
      {
        "type": "value",
        "parameter": {
          "source": "payload",
          "name": "payload.event"
        },
        "value": "media.play"
      }
    }
  }
]

```

Thanks for helping!

Eric

**Reply by moorereason** [reply]
You'll need to update webhook to at least v2.7.0 to work with Plex.  Same issue was reported in https://github.com/adnanh/webhook/issues/311.

**Reply by elacoursiere** [reply]
Solved, thanks!! 💯
